angular/angular-checkout/src/app/modules/checkout/checkout.component.ts

```diff
@@ -175,6 +242,57 @@ export class CheckoutComponent implements OnInit, OnDestroy {
     return response.json();
   }
 
+  private async createCustomerAndAttachPM(paymentMethodId: string): Promise<void> {
+    const customerName =
+      this.paymentMethodType === 'card'
+        ? this.cardPaymentForm.get('cardholderName').value
+        : this.achDebitPaymentForm.get('accountholderName').value || '';
+    const customerDetails = {
+      first_name: customerName.split(' ')[0],
+      last_name: customerName.split(' ')[1] || '',
+      metadata: {
+        internal_customer_id: '12345', // You can set logic to generate a unique customer id or use the id from your system
+      },
+    };
+
+    const requestHeaders: HeadersInit = new Headers();
+    requestHeaders.set('Content-Type', 'application/json');
+    if (this.merchantAccountId) requestHeaders.set('tilled_account', this.merchantAccountId);
+    const customerResponse = await fetch('/api/customer', {
+      method: 'POST',
+      headers: requestHeaders,
+      body: JSON.stringify(customerDetails),
+    });
+    if (!customerResponse.ok) {
+      throw new Error(`Unable to create customer and payment method. Status: ${customerResponse.statusText}`);
+    }
+
+    const customer = await customerResponse.json();
+    const pmAttach = await fetch(`/api/payment-methods/${paymentMethodId}/attach`, {
+      method: 'PUT',
+      headers: requestHeaders,
+      body: JSON.stringify({
+        customer_id: customer.id,
+      }),
+    });
+    if (!pmAttach.ok) {
+      throw new Error(`Unable to attach payment method to customer. Status: ${pmAttach.statusText}`);
+    }
+    const pmData = await pmAttach.json();
+    this.receiptDetails = {
+      merchantName: this.merchantName,
+      paymentMethodDetails:
+        pmData.type === 'card'
+          ? pmData.card.brand.toUpperCase() + ' ' + pmData.card.last4
+          : pmData.ach_debit.account_type.charAt(0).toUpperCase() +
+            pmData.ach_debit.account_type.slice(1) +
+            ' **' +
+            pmData.ach_debit.last2,
+      customerName: customer.first_name + ' ' + (customer.last_name || ''),
+    };
+    this.showReceipt = true;
```

Comment:
You probably need some error handling before you show the receipt to show an error message if the payment fails instead of displaying a broken reciept
![Screenshot 2024-06-20 at 10 37 57 AM](https://github.com/gettilled/tilled-example-monorepo/assets/65815343/464d0e4d-5f03-4009-b878-0c8b2ce5a554)

Reply:
You could also add input validation on the zip field to limit it to 10characters